dockstore-webservice/src/main/java/io/dockstore/webservice/resources/UserResource.java

```diff
@@ -435,15 +438,17 @@ public User removeUserFromGroup(@ApiParam(hidden = true) @Auth User authUser,
         return FluentIterable.from(byId.getEntries()).filter(Tool.class).toList();
     }
 
-
     @GET
     @Timed
     @UnitOfWork
     @Path("/starredTools")
     @ApiOperation(value = "Get the logged-in user's starred tools", response = Entry.class, responseContainer = "List")
     public Set<Entry> getStarredTools(@ApiParam(hidden = true) @Auth User user) {
         User u = userDAO.findById(user.getId());
-        return u.getStarredEntries().stream().filter(element -> element instanceof Tool).collect(Collectors.toSet());
+        Comparator<Entry> byEntryId = Comparator.comparingLong(Entry::getId);
```

Comment:
Just sort using a JPA annotation such as 
https://docs.oracle.com/javaee/7/api/javax/persistence/OrderBy.html

This ensures that all code that uses getStarredEntries will get a sorted collection rather than fixing each method independently. Also, best to add a test for this kind of change.